Lib/datetime.py

```diff
@@ -1687,13 +1770,15 @@ def utcfromtimestamp(cls, t):
     @classmethod
     def now(cls, tz=None):
         "Construct a datetime from time.time() and optional time zone info."
-        t = _time.time()
+        t = _time.time_ns()
+        t = _Decimal(t) / _Decimal(1e9)
```

Comment:
Again, unsafe. It may be worthwhile making an internal version of `fromtimestamp` that expects the argument as an `int` number of nanoseconds, and implementing all other uses (including `fromtimestamp`) in terms of that, removing the need for high precision decimal values entirely.